STYLE_GUIDE.md

```diff
@@ -0,0 +1,48 @@
+# Style guide
+
+This style guide defines specific coding standards and advice for this Java codebase. The rules here are extensions to the [Google Java Style Guide](https://google.github.io/styleguide/javaguide.html).
+
+Please see the [contributing guide](CONTRIBUTING.md) for general guidance for contributing to this project.
+
+### Automatic formatting
+
+Automatic formatting should be performed with `./gradlew goJF` or `./mvnw fmt:format`. Formatting all projects can be done with `./build.sh format`.
+
+### Class member order
+
+*Extends [3.4.2](https://google.github.io/styleguide/javaguide.html#s3.4.2-ordering-class-contents)*
+
+Class members should be in the following order, in decreasing priority:
+
+1. Static before non-static
+1. Nested classes/interfaces before fields before constructors before methods
+1. Public before private
+1. Final before non-final
+
+### Public APIs
+
+User-facing methods (such as those in Jib Core) should not have types in their signature that are not standard JDK classes. For example, a parameter should take type `List` rather than Guava's `ImmutableList`.
+
+### Package hierarchy
+
+Packages should depend on each other without cycles.
```

Comment:
you mean java packages? what about sub-packages? This is a little bit more stringent than what good build systems enforce which is no-cycles between build level modules.